tests/suites/test_suite_shax.function

```diff
@@ -109,6 +110,292 @@ void mbedtls_sha512(char *hex_src_string, char *hex_hash_string )
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_SHA3_C */
+void mbedtls_sha3_224( char *hex_src_string, char *hex_hash_string )
+{
+    unsigned char src_str[10000];
+    unsigned char hash_str[10000];
+    unsigned char output[129];
+    int src_len;
+
+    memset(src_str, 0x00, 10000);
+    memset(hash_str, 0x00, 10000);
+    memset(output, 0x00, 129);
+
+    src_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_sha3( src_str, src_len, MBEDTLS_SHA3_224, output);
+    hexify( hash_str, output, 28 );
+
+    TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SHA3_C */
+void mbedtls_sha3_256( char *hex_src_string, char *hex_hash_string )
+{
+    unsigned char src_str[10000];
+    unsigned char hash_str[10000];
+    unsigned char output[129];
+    int src_len;
+
+    memset(src_str, 0x00, 10000);
+    memset(hash_str, 0x00, 10000);
+    memset(output, 0x00, 129);
+
+    src_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_sha3( src_str, src_len, MBEDTLS_SHA3_256, output);
+    hexify( hash_str, output, 32 );
+
+    TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SHA3_C */
+void mbedtls_sha3_384( char *hex_src_string, char *hex_hash_string )
+{
+    unsigned char src_str[10000];
+    unsigned char hash_str[10000];
+    unsigned char output[129];
+    int src_len;
+
+    memset(src_str, 0x00, 10000);
+    memset(hash_str, 0x00, 10000);
+    memset(output, 0x00, 129);
+
+    src_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_sha3( src_str, src_len, MBEDTLS_SHA3_384, output);
+    hexify( hash_str, output, 48 );
+
+    TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SHA3_C */
+void mbedtls_sha3_512( char *hex_src_string, char *hex_hash_string )
```

Comment:
can we combine `mbedtls_sha3_512` , `mbedtls_sha3_384`, `mbedtls_sha3_256` and `mbedtls_sha3_224` to one function `mbedtls_sha3` that will also receive `mbedtls_sha3_type_t ` and output size? duplicating the same function with only two slightly different values is redundant.

Reply:
Good point. Daniel followed what Paul did for SHA-2, but I can't see a good reason for that.

Reply:
Done in “Simplfy SHA-3 test code”